src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -28,11 +28,18 @@ public static IConfigurationBuilder AddUserSecrets<T>(this IConfigurationBuilder
             => configuration.AddUserSecrets(typeof(T).GetTypeInfo().Assembly);
 
         /// <summary>
+        /// <para>
+        /// This method is obsolete and will be removed in a future version.
+        /// The recommended alternative is <see cref="AddUserSecrets(IConfigurationBuilder, string)"/>.
+        /// </para>
+        /// <para>
         /// Adds the user secrets configuration source. Searches the assembly from <see cref="Assembly.GetEntryAssembly"/>
         /// for an instance of <see cref="UserSecretsIdAttribute"/>.
+        /// </para>
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is .AddUserSecrets(string userSecretsId).")]
```

Comment:
Let's compromise. Since we have shortened the message to "calling" style, how about `...the recommended alternative is .AddUserSecrets<TStartup>() or .AddUserSecrets(string userSecretId)`